src/components/Common/FilePreviewDialog.tsx

```diff
@@ -171,12 +209,104 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
   };
 
   useKeyboardShortcut(["ArrowLeft"], () => index > 0 && handleNext(index - 1));
-
   useKeyboardShortcut(
     ["ArrowRight"],
     () => index < (uploadedFiles?.length || 0) - 1 && handleNext(index + 1),
   );
 
+  const handleMouseDown = (e: React.MouseEvent) => {
+    if (!file_state.isImage) return;
+    setDragState((prev) => ({
+      ...prev,
+      isDragging: true,
+      dragStart: {
+        x: e.clientX - prev.position.x,
+        y: e.clientY - prev.position.y,
+      },
+    }));
+  };
+
+  const handleMouseMove = (e: React.MouseEvent) => {
+    if (!dragState.isDragging) return;
+    const container = e.currentTarget as HTMLDivElement;
+    const image = container.querySelector("img");
+    if (!image) return;
+
+    const containerRect = container.getBoundingClientRect();
+    const imageRect = image.getBoundingClientRect();
+
+    const maxX = Math.max(0, (imageRect.width - containerRect.width) / 2);
+    const maxY = Math.max(0, (imageRect.height - containerRect.height) / 2);
+
+    const newX = e.clientX - dragState.dragStart.x;
+    const newY = e.clientY - dragState.dragStart.y;
+
+    const clampedX = Math.max(-maxX, Math.min(maxX, newX));
+    const clampedY = Math.max(-maxY, Math.min(maxY, newY));
+
+    setDragState((prev) => ({
+      ...prev,
+      position: {
+        x: clampedX,
+        y: clampedY,
+      },
+    }));
+  };
+
+  const handleMouseUp = () => {
+    setDragState((prev) => ({
+      ...prev,
+      isDragging: false,
+    }));
+  };
+
+  const handleTouchStart = (e: React.TouchEvent) => {
+    if (!file_state.isImage) return;
+    setDragState((prev) => ({
+      ...prev,
+      isDragging: true,
+      dragStart: {
+        x: e.touches[0].clientX - prev.position.x,
+        y: e.touches[0].clientY - prev.position.y,
+      },
+    }));
+  };
+
+  const handleTouchMove = (e: React.TouchEvent) => {
```

Comment:
Isn't the logic exact same as handleMouseMove? handle it accordingly.

Reply:
➕